### DISCUSSION TITLE
Anvil seems not to be compatible with returning data from the constructor of a read-only deployment eth_call

### DISCUSSION THREAD

**Discussion by Gilgames000** [foundry-rs/foundry#4549] [General]
I've been using [a library](https://github.com/Destiner/ethcall) that supports [deployless multicall](https://destiner.io/blog/post/deployless-multicall/) on networks where the contract is not deployed or if you try to fetch historical state at a block number where the multicall contract was not available. It works with all the nodes I tried (Infura, Alchemy, Tenderly, Hardhat, etc.) except Anvil, where it seems to return `0x` as the data.

From the [deployless multicall Solidity implementation repo](https://github.com/Destiner/deployless-multicall) :
> Normally, a consturcor in Solidity contract returns the bytecode of the newly created contract. Here, we use inline assembly to overwrite return data with the call result.
> 
> The code makes some assuptions about how the variables stored in memory, so it's better to stick to Solidity version specified in the contract.

Is this intended behavior in Anvil?

**Comment by 0xcoinymous** [comment]
I have the same issue. I deployed my contract on 9 chains. only eth_call to Anvil returns `0x`
